account_cancel_with_reversal/tests/test_account_move_reverse.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from openerp.tests import common
+
+
+class TestAccountInvoiceCancel(common.TransactionCase):
+    """
+    Case1 - test_invoice_cancel:
+    - Create draft invoice and validate and Cancel
+    - When invoice cancel reverse entry should be generated.
+    Case2 - test_voucher_cancel:
+    - Create draft voucher and validate and Unreconcile
+    - When voucher Unreconcile reverse entry should be generated.
+    """
+    def setUp(self):
+        super(TestAccountInvoiceCancel, self).setUp()
+        cr, uid = self.cr, self.uid
+        self.account_invoice_model = self.registry('account.invoice')
+        self.account_invoice_line_model = self.registry('account.invoice.line')
+        self.wizard_voucher_move_reverse = \
+            self.registry('account.move.reverse')
+        self.account_voucher_model = self.registry('account.voucher')
+        self.model_data = self.registry("ir.model.data")
+
+        self.partner_id = self.model_data.get_object_reference(
```

Comment:
For reference object in v8, you can use,
`self.partner = self.env.ref('base.res_partner_2')`
Note, that it is object, so I use `self.partner`, not `self.partner_id`
Using v8 API this way, your code will also be shorter as it need no browse()